### PR TITLE
Relational repository for Sample Users & Orders

### PR DESCRIPTION
**Pull request by DeanFaizal**

### CODE REVIEW

File: in_memory_repository/readme.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ This repository contains sample use cases for an in memory repository implementa
   
 ## Diagram
 Here is a diagram of how the sample implements the MobCAT repository pattern:
-
+![repository pattern graphic](repository_pattern_graphic.jpg)
```
Comment (BenBtg) [inline in_memory_repository/readme.md]: Nice!

File: in_memory_repository/readme.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+2. The second sample shows how to fetch relational items with queries using the InMemorySampleRepositoryContext. The items are:
+  - SampleUserModel
+  - SampleOrderModel. Each order is tied to a User through the UserId Property.
+  
+## Diagram
+Here is a diagram of how the sample implements the MobCAT repository pattern:
+
+
+## Reasoning/Rationale
+The pattern exposes high level methods through the InMemorySampleRepositoryContext which impelments IOfflineSampleRepositoryContext. To implement a different repository type (SQLite, EFCore, Akavache, etc), all it would need to do is implement IOfflineSampleRepositoryContext and be registered in App.xaml.cs.
```
Comment (BenBtg) [inline in_memory_repository/readme.md]: implements

File: in_memory_repository/readme.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+The reasoning behind the RepositoryContext is to avoid having to cast objects directly from the Business Logic type to the underlying implementation type. All the translation is done at the repository level, providing better encapsulation and separation of reponsibilities. This also allows the business logic models to be totally independent of the repository models.
+
+From the perspective of the app, this single point of entry to the repository provides a simple way to mock the repository for different environments, demos, QA testing, and automated testing.
+
+From the perspective of the library, this reduces complexity for the consumer. All the consumer needs to do is the RepositoryContext interface directly and swap out the dependency service registration.
```
Comment (BenBtg) [inline in_memory_repository/readme.md]: "All the consumer needs to do is the "
implement?

File: in_memory_repository/OfflineSample/OfflineSample.csproj [hunk 4]
```diff
@@ -18,8 +18,11 @@
   </ItemGroup>
 
   <ItemGroup>
-    <Reference Include="MobCAT">
```
Comment (BenBtg) [inline in_memory_repository/OfflineSample/OfflineSample.csproj]: Need to update Forms to 4.4

File: in_memory_repository/OfflineSample/Services/ItemGeneratorService.cs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xamarin.Forms;
+
+namespace OfflineSample.Services
+{
+    public class ItemGeneratorService : IItemGeneratorService
```
Comment (BenBtg) [inline in_memory_repository/OfflineSample/Services/ItemGeneratorService.cs]: Great idea! Saves a lot of time to demo.

File: in_memory_repository/OfflineSample.Data.InMemory/Repositories/InMemorySampleOrderRepository.cs [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+        public async Task<IEnumerable<SampleOrderModel>> GetOrdersAsync(string userId)
+        {
+            var ordersForUser = await ExecuteTableQueryAsync(a => a.UserId == userId);
+            return ordersForUser?.Select(a => ToModelType(a));
+        }
+
+        public async Task<IEnumerable<SampleOrderModel>> GetOrdersWithMinimumCountAsync(int minimumCount)
+        {
+            var orders = await GetAsync();
```
Comment (mikeparker104) [inline in_memory_repository/OfflineSample.Data.InMemory/Repositories/InMemorySampleOrderRepository.cs]: We should use ExecuteTableQueryAsync to perform the query directly against the database without first retrieving all records

**Pull request merged**